### DISCUSSION TITLE
Question about MSB and LSB in PS1 SEQ files

### DISCUSSION THREAD

**Discussion by Meerkov** [vgmtrans/vgmtrans#293] [Music formats]
![image](https://user-images.githubusercontent.com/8139496/151688172-1b2ec90d-a7b5-438e-a7b8-15219ea9d721.png)
In the above PS1 SEQ, there is a short section of dark gray. These are indicated as "NRPN 2" and "NRPN 1" corresponding to MSB and LSB.

From what I've read in LibOver47.pdf, it reads like this:
00 BF 63 28 (Mark on the MSB- but I don't know what Mark means, and not sure what the extra BF does)
00 62 60 (Mark 60 on the LSB)
00 63 28 (Mark on the MSB)
00 62 00 (Mark 0)
then it repeats this sequence several more times, alternating, with the only change being that the LSB will instead mark 
61, 0, 62, 0, 63, 0, 64, 0, 65, 0, 66, 1, 67 0, and finally 41...

I don't really know how to interpret what this is intended for. It doesn't seem to play as a midi. Indeed, it has no notes... So what is all this marking business? What is this file intended to do?

**Comment by loveemu** [comment]
In conclusion I do not understand the intent of this sequence, but I can answer a few things.

The basic understanding is correct: in the NRPN message of SEQ, the NRPN (CC99/CC98) represents the operation, the data entry (CC6) represents the parameter for the operation. And according to the library reference, NRPN MSB=40 represents "mark". This function is used to call the callback function set in SsSetMarkCallback, so it is up to the callback implementation how the program reacts to it.

What's odd is that there is no data entry (CC6) anywhere. In general, I don't think any event will occur until the data entry comes in. If the game is using a custom driver similar to libsnd, it is not inconceivable that these could be completely different events from the markers.

Oh, and 0xBF is a message byte indicating that it is a control change on the 15th channel. Since the running status rule applies, this value is omitted for the second and subsequent control changes.

By the way, this was my first time to actually see the SEQ header stored in the SEP. What game/file is this? Just curious.

**Reply by Meerkov** [reply]
This is from 3Xtreme.

Each .SND file in the iso has 3 SEQ files in it, 2 VABs, and 1 Instrument Coll.

The three SEQ files are: One Control Sequence (Like the one above, that is doing unknown something), then an Introduction (usually the first couple seconds to a minute of the song), then the last file is the rest of the song (presumably, the part that repeats).

So my best guess is that the very short entry above is doing some kind of setup for the song, but it's just a guess.

Edit: My understanding of why you don't usually see the pQES header is that is only appears on the first entry in the SEP file. All other entries include garbage data due to hacky parsing in VGMTrans (There's this weird `+x13 - 6`  offset which tries to align the parser without rewriting anything, so it captures the end of the previous song as the header of the next song). Luckily, afaik, that data is unused anyway.
